source/guides/clear/performance.rst

```diff
@@ -0,0 +1,236 @@
+manage this complexity and to apply the techniques used in the software build
+toolchain across the entire project. autospec is available as part of the OS for
+developers to use when they build their own projects on |CL|.
+
+Latest versions of compilers and low-level libraries
+====================================================
+
+|CL| is a rolling release distribution and follows upstream software
+repositories, including compilers and libraries, for updates. |CL| includes
+upstream source-level optimizations as soon as they're available.
+
+A benchmark approach to compiler performance
+============================================
+
+|CL| chooses the compiler used to build each software package on a case-by-case
+basis to maximize performance. Typically, |CL| uses the open source `GNU Compiler
+Collection <https://gcc.gnu.org/>`_ (GCC) which includes the standard low-level
+libraries: `Glibc <https://www.gnu.org/software/libc/>`_ / 
+`libstdc++ <https://gcc.gnu.org/onlinedocs/libstdc++/>`_ for C and C++
+programming languages. If there is a performance advantage, |CL| will build
+packages with `Clang / LLVM <https://clang.llvm.org/>`_.
+
+|CL| uses patched compilers and low-level libraries for exact control of the
+software build. Patches include changes that default to more aggressive
+optimizations or optimizations that haven't yet been merged upstream.
+
+View the full list of patches in the autospec repositories on GitHub:
+
+* https://github.com/clearlinux-pkgs/gcc
+* https://github.com/clearlinux-pkgs/glibc
+* https://github.com/clearlinux-pkgs/llvm
+
+Aggressive compiler flags
+=========================
+
+|CL| uses aggressive
+`compiler flags <https://gcc.gnu.org/onlinedocs/gcc/Optimize-Options.html>`_ to
+optimize software builds for runtime performance. Some significant flags that
+|CL| often implements are:
+
+`mtune and march <https://gcc.gnu.org/onlinedocs/gcc/x86-Options.html>`_
+	Options used to tune generated code with optimized instructions for specific
+	CPU types instead of creating generic code for maximum compatibility.
+
+	|CL| defines it's minimum hardware requirements to be second-generation
+	Intel® microarchitecture code name Westmere (released in 2010) or later.
+	This enables the OS to build on newer CPU architecture features. Whenever
+	possible, |CL| tunes code for the Haswell generation processors or newer.
+
+	|CL| sets	:command:`march=westmere` and :command:`mtune=haswell`.
+
+	.. note::
+		|CL| doesn't require Advanced Encryption Standard (AES), so it may run
+		on some Intel CPUs from the first generation of Intel® microarchitecture
+		code name Nehalem (released in 2008). Refer to the
+		`recommended minimum system requirements <https://docs.01.org/clearlinux/latest/reference/system-requirements.html>`_ for specific requirements.
+
+`O3 <https://gcc.gnu.org/onlinedocs/gcc/Optimize-Options.html>`_
+	The largest preset of compiler options optimizations for performance. O3
+	favors runtime performance.

```

Comment:
perhaps a footnote here is in order - `man gcc -> "Optimization Options".

Reply:
Added.